### PR TITLE
feat: Backend Validation for Creation Method

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY
This creates backend validation to make sure that a user cannot have multiple reports attached to one chart or dashboard with the respective creation method. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: tests/integration_tests/reports/api_tests.py [hunk 1]
```diff
@@ -790,6 +790,96 @@ def test_no_dashboard_report_schedule_schema(self):
...
+            "description": "description",
+            "creation_method": ReportCreationMethodType.CHARTS,
+            "crontab": "0 9 * * *",
+            "working_timeout": 3600,
+            "chart": chart.id,
+        }
+        uri = "api/v1/report/"
+        rv = self.client.post(uri, json=report_schedule_data)
+        data = json.loads(rv.data.decode("utf-8"))
+        assert rv.status_code == 422
```
Comment (betodealmeida) [inline tests/integration_tests/reports/api_tests.py:833]: I think returning a 409 (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409) makes more sense here. All you need to do is to set the attribute `status = 409` in your new exception.

File: superset/reports/commands/exceptions.py [hunk 2]
```diff
@@ -167,6 +167,19 @@ def __init__(self) -> None:
...
 
 
+class ReportScheduleCreationMethodUniquenessValidationError(ValidationError):
+    """
+    Marshmallow validation error for Report Schedule with creation method charts
+    or dashboards already existing on a report.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(
```
Comment (betodealmeida) [inline superset/reports/commands/exceptions.py:177]:
```suggestion
        status = 409
        
        super().__init__(
```

**Review by betodealmeida — CHANGES_REQUESTED** [review]

File: tests/integration_tests/reports/api_tests.py (hunk 1 above)
Comment (AAfghahi) [reply tests/integration_tests/reports/api_tests.py]: changed

**Review by betodealmeida — APPROVED** [review]
Looks great!

File: superset/reports/commands/exceptions.py [hunk 3]
```diff
@@ -167,6 +167,21 @@ def __init__(self) -> None:
...
 
 
+class ReportScheduleCreationMethodUniquenessValidationError(ValidationError):
+    """
+    Marshmallow validation error for Report Schedule with creation method charts
+    or dashboards already existing on a report.
+    """
+
+    def __init__(self) -> None:
+        status = 409
```
Comment (betodealmeida) [inline superset/reports/commands/exceptions.py:177]: Sorry, this should be on the class, not inside `__init__`:

```
class ReportScheduleCreationMethodUniquenessValidationError(ValidationError):
    status = 409

    def __init__(self) -> None:
       super().__init__(...)
```

**Pull request merged**